### ISSUE TITLE
[no-unnecessary-condition] False positive from && or || within JSX expression container

### ISSUE THREAD

**Issue by threehams** [typescript-eslint/typescript-eslint#979]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "error",
  }
}
```

```tsx
const Component = () => {
  const maybeString: string | undefined = "something";
  return (
    <>
      { maybeString && <div>{maybeString}</div>}
    </>
  )
}
```

**Expected Result**
Error: "Unnecessary conditional, value is always truthy."

**Actual Result**
No error.

**Additional Info**
This appears to affect both `||` and `&&` inside JSX

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.2.1-alpha.8` |
| `@typescript-eslint/parser`        | `2.2.1-alpha.8` |
| `TypeScript`                       | `3.6.3` |
| `ESLint`                           | `6.4.0` |
| `node`                             | `12.8.0` |
| `npm`                              | `6.10.2` |

**Comment by threehams** [comment]
Not a bug, it looks like I'm needing the `ignoreRhs` option.
https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-unnecessary-condition.md#options

I'm surprised this isn't the default, but I'm not sure how common this is outside of React.

**Comment by bradzacher** [comment]
It's not the default because it was added after the rule was released.
Adding an option is a minor version bump.
Adding an option and turning it on by default is a breaking change.
